src/library/Utilities/RemoteContainerConnectionDetailsResolver.cs

```diff
@@ -49,6 +49,17 @@ internal class RemoteContainerConnectionDetailsResolver
                 return remoteContainerConnectionDetails;
             }
 
+            // Check if restoration pod is present. This is an indication that previous session is still connected or it has not finished restoring yet.
+            var allPods = (await _kubernetesClient.ListPodsInNamespaceAsync(remoteContainerConnectionDetails.NamespaceName, null, cancellationToken))?.Items;
+            var podsRunningRestorationJob = allPods.Where(p => p.Metadata.Name.StartsWith(remoteContainerConnectionDetails.ServiceName + "-restore") &&
+                                                        (p.Status?.ContainerStatuses?.Where(cs => cs.Image.Contains(ImageProvider.DevHostRestorationJob.Name)).Any() ?? false));
+            if (podsRunningRestorationJob.Count() > 0)

```

Comment:
💡☕️🙏 Just idea: Can this ever be null? if so might be check if this function is in utils already.

```.cs
public static bool IsNullOrEmpty(this Array array)
{
    return (array == null || array.Length == 0);
}

Reply:
FYI - Dotnet framework has a string.IsNullOrEmpty() method which can be used here.

Reply:
podsRunningRestorationJob will never be null, it will just be an empty list if no matches. Let me know if you were refereing to a variable different than podsRunningRestorationJob.

Reply:
Cool, sounds good ☕️👍 I mentioned because: if you check in line 53 where new var allpods is defined it uses “?” Nullable type and then at line 54 this var.

Suggestion is - Since you might know the scenario feel feee to use null-forgiving operator where null needs suppressing. 

Please feel free to resolve this conversation if not needed, thanks.

Reply:
Thank @Tatsinnit! I understand your comment now. This is a good catch. I did not find function static bool IsNullOrEmpty(this Array array) so I am just doing null check